### PR TITLE
[NavigationView]: Fix bug where footer scrollbar would still show up on v1 style

### LINKED ISSUE #3771: [NavigationView] FooterMenuItems show unwanted scrollbar
**Describe the bug**
NavigationView FooterMenuItems show unwanted scrollbar

**Steps to reproduce the bug**
1. Add a `NavigationView `and add some `NavigationViewItem`s
2. Hover with the mouse over the right area of the FooterMenuItems area

**Expected behavior**
No Scrollbar when there's enough space

**Screenshots**
image

NuGet package version: 
Microsoft.UI.Xaml 2.5.0

**Additional context**
Might have been introduced here: https://github.com/microsoft/microsoft-ui-xaml/issues/3447

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Fixed this by backporting changes to v1 style. The bug only occured in V1 styles!
## Motivation and Context

Closes #3771
## How Has This Been Tested?

Tested manually, still works as expected

### CODE REVIEW

File: dev/NavigationView/NavigationView_v1.xaml [hunk 1]
```diff
@@ -493,6 +493,7 @@
                                             <!-- FooterItems -->
                                             <local:ItemsRepeaterScrollHost Grid.Row="3">
                                                 <ScrollViewer x:Name="FooterItemsScrollViewer"
+                                                        VerticalScrollBarVisibility="Auto"
```
Comment (beervoley) [inline dev/NavigationView/NavigationView_v1.xaml:496]: @chingucoding I think the only change that's needed to hide the scrollbar is this line. I'm thinking we shouldn't touch the rest of the code? Please, correct me if I'm missing something:)

Comment (chingucoding) [reply dev/NavigationView/NavigationView_v1.xaml:496]: There are three reasons why I think that changing the other lines is good:
- They are outdated and use names that aren't used anywhere
- The templates are diverging for no good reason making them harder to maintain when looking at the layout code for example
- There was a visual bug where the footers would have been cut off

**Review by beervoley — APPROVED** [review]

**Comment by beervoley** [comment]
@karenbtlai could you have a look please?:)

**Pull request merged**